src/Analysis/Ast/Test/AssignmentTests.cs

```diff
@@ -308,6 +308,48 @@ class C:
                 .Which.Should().HaveMembers(listMemberNames);
         }
 
+        [TestMethod, Priority(0)]
+        public async Task AssignBeforeClassDef() {
+            const string code = @"
+D = 5
+class D: ...
+";
+            var analysis = await GetAnalysisAsync(code);
+            analysis.Should().HaveVariable("D").OfType("D");
+        }
+
+        [TestMethod, Priority(0)]
+        public async Task AssignAfterClassDef() {
+            const string code = @"
+class D: ...
+D = 5
+";
+            var analysis = await GetAnalysisAsync(code);
+            analysis.Should().HaveVariable("D").OfType(BuiltinTypeId.Int);
+        }
+
+        [TestMethod, Priority(0)]
+        public async Task AssignBeforeFunctionDef() {
+            const string code = @"
+D = 5
+def D():
+    pass
+";
+            var analysis = await GetAnalysisAsync(code);
+            analysis.Should().HaveVariable("D").OfType("D");
```

Comment:
Is "having D of type D" the right way to check that `D` is a function? Seems weird to me.

Reply:
Ok